poh/src/poh_recorder.rs

```
@@ -602,6 +605,37 @@ impl PohRecorder {
             for (entry, tick_height) in &self.tick_cache[..entry_count] {
                 working_bank.bank.register_tick(&entry.hash);
 
+                if is_alpentick {
+                    // Wait for the bank to be frozen with timeout
+                    let start = Instant::now();
+                    while !working_bank.bank.is_frozen() {
+                        if start.elapsed() > Duration::from_secs(1) {
+                            break;
+                        }
+                        std::hint::spin_loop();
+                    }
+
+                    // If the bank still isn't frozen, we've timed out
+                    if !working_bank.bank.is_frozen() {
+                        break;
```

is this recoverable?

---

sadly no - at this point, we've timed out, and in practice, the leader's full window ends up getting skipped

will add logging here though so we don't silently error

---

Log is definitely good.

But by recoverable I meant can the validator continue to function after this? Like will it be able to validate blocks and maybe even produce its next leader window?

If yes, error log and break seems good.

If not, would rather log and shut down the validator.

---

Do we need to worry about some poisonous transaction sneaking in which drags replay on forever here?